src/components/AudioPlayer/AudioPlayer.web.js

```diff
@@ -39,24 +39,41 @@ export default class AudioPlayerSub extends Component {
 
   componentDidUpdate(prevProps) {
     const {
-      track: { url },
+      track: { url, title: title },
       playing,
       progress,
-      endSong,
+      updatePlaying,
+      topScreen,
     } = this.props
     const {
-      track: { url: prevUrl },
+      track: { url: prevUrl, title: prevTitle },
       playing: prevPlaying,
     } = prevProps
+
     if (url !== prevUrl) {
       this.addUrl(url)
     }
+
     if (playing !== prevPlaying) {
       if (playing && this.player.src) this.player.play()
       else if (!playing && this.player.src) this.player.pause()
     }
-    // Check if the song has ended, and if so call the endSong function from index
-    if (Math.round(progress * 100) / 100 === 1) endSong()
+
+    // If song has ended, reset progress and trigger end action
+    if (topScreen && Math.round(progress * 10000) / 10000 === 1) {
+      const { updatePlayed, updateProgress, endSong } = this.props
+      this.player.currentTime = 0
+      updatePlayed(0)
+      updateProgress(0)
+
+      endSong()
+    }
+
+    // Pauses audio when navigating to different screen
+    if (!topScreen && playing) {
```

Comment:
You probably want to add the automatic pausing logic here too right?

Reply:
I posted in the marketplace channel about this, so let me know what you think. Right now we're gonna switch auto-pause back to in-app background audio or something like that, then add help text to say that you can only toggle in-app background audio on mobile.